src/test/scala/com/microsoft/hyperspace/goldstandard/PlanStabilitySuite.scala

```diff
@@ -0,0 +1,289 @@
+      FileUtils.deleteDirectory(dir)
+      assert(dir.mkdirs())
+
+      val file = new File(dir, "simplified.txt")
+      FileUtils.writeStringToFile(file, simplified, StandardCharsets.UTF_8)
+      val fileOriginalPlan = new File(dir, "explain.txt")
+      FileUtils.writeStringToFile(fileOriginalPlan, explain, StandardCharsets.UTF_8)
+      logDebug(s"APPROVED: $file $fileOriginalPlan")
+    }
+  }
+
+  private def checkWithApproved(plan: SparkPlan, name: String, explain: String): Unit = {
+    val dir = getDirForTest(name)
+    val tempDir = FileUtils.getTempDirectory
+    val actualSimplified = getSimplifiedPlan(plan)
+    val foundMatch = isApproved(dir, actualSimplified)
+
+    if (!foundMatch) {
+      // show diff with last approved
+      val approvedSimplifiedFile = new File(dir, "simplified.txt")
+      val approvedExplainFile = new File(dir, "explain.txt")
+
+      val actualSimplifiedFile = new File(tempDir, s"$name.actual.simplified.txt")
+      val actualExplainFile = new File(tempDir, s"$name.actual.explain.txt")
+
+      val approvedSimplified =
+        FileUtils.readFileToString(approvedSimplifiedFile, StandardCharsets.UTF_8)
+      // write out for debugging
+      FileUtils.writeStringToFile(actualSimplifiedFile, actualSimplified, StandardCharsets.UTF_8)
+      FileUtils.writeStringToFile(actualExplainFile, explain, StandardCharsets.UTF_8)
+
+      fail(s"""
+          |Plans did not match:
+          |last approved simplified plan: ${approvedSimplifiedFile.getAbsolutePath}
+          |last approved explain plan: ${approvedExplainFile.getAbsolutePath}
+          |
+          |$approvedSimplified
+          |
+          |actual simplified plan: ${actualSimplifiedFile.getAbsolutePath}
+          |actual explain plan: ${actualExplainFile.getAbsolutePath}
+          |
+          |$actualSimplified
+        """.stripMargin)
+    }
+  }
+
+  /**
+   * Get the simplified plan for a specific SparkPlan. In the simplified plan, the node only has
+   * its name and all the sorted reference and produced attributes names(without ExprId) and its
+   * simplified children as well. And we'll only identify the performance sensitive nodes, e.g.,
+   * Exchange, Subquery, in the simplified plan. Given such a identical but simplified plan, we'd
+   * expect to avoid frequent plan changing and catch the possible meaningful regression.
+   */
+  private def getSimplifiedPlan(plan: SparkPlan): String = {
+    val exchangeIdMap = new mutable.HashMap[SparkPlan, Int]()
+    val subqueriesMap = new mutable.HashMap[SparkPlan, Int]()
+
+    def getId(plan: SparkPlan): Int = plan match {
+      case exchange: Exchange => exchangeIdMap.getOrElseUpdate(exchange, exchangeIdMap.size + 1)
+      case ReusedExchangeExec(_, exchange) =>
+        exchangeIdMap.getOrElseUpdate(exchange, exchangeIdMap.size + 1)
+      case subquery: SubqueryExec =>
+        subqueriesMap.getOrElseUpdate(subquery, subqueriesMap.size + 1)
+      case _ => -1
+    }
```

Comment:
Note: removed a couple of `case ` match statements because of lack of spark 3.0 support
```scala
case SubqueryBroadcastExec
case ReusedSubqueryExec
```